src/env/as_env.rs

```diff
@@ -81,14 +81,38 @@ pub fn assembly_script_abort(
     let message = message.read(memory);
     let filename = filename.read(memory);
     if message.is_err() || filename.is_err() {
-        abi_bail!("Aborting failed to load message or filename")
+        abi_bail!("aborting failed to load message or filename")
     }
-    eprintln!(
-        "Error: {} at {}:{} col: {}",
+    abi_bail!(format!(
+        "error: {} at {}:{} col: {}",
         message.unwrap(),
         filename.unwrap(),
         line,
         col
-    );
-    Ok(())
+    ));
+}
+
+/// Assembly script builtin export `seed` function
+pub fn assembly_script_seed(env: &ASEnv) -> ABIResult<f64> {
+    match env.interface.unsafe_random() {
+        Ok(ret) => Ok(ret as f64),
```

Comment:
this converts a random `i64`into a random `f64`. This is problematic as it covers a very small space of f64:
```
i64 max = 9_223_372_036_854_775_807
i64 min = - 9_223_372_036_854_775_808
f64 max = 1.7976931348623157E+308
f64 min = - 1.7976931348623157E+308
```

We should add another interface function to generate a random f64 using:

```
    fn unsafe_random_f64(&self) -> Result<f64> {
        let distr = rand::distributions::Uniform::new_inclusive(f64::MIN, f64::MAX);
        Ok(context_guard!(self).unsafe_rng.sample(distr))
    }
```